src/blockchain_db/sqlite/db_sqlite.cpp

```diff
@@ -147,12 +156,57 @@ void BlockchainSQLite::load_database(std::optional<fs::path> file)
       SQLite_busy_timeout.count()
       });
 
-  if (!db->tableExists("batch_sn_payments")) {
+  if (!db->tableExists("batch_sn_payments") || !db->tableExists("batch_sn_info")) {
     create_schema();
   }
+
+  SQLite::Statement st{*db, "SELECT height FROM batch_sn_info"};
+  while (st.executeStep()) {
+    this->height = st.getColumn(0).getInt();
+  }
+}
+
+bool BlockchainSQLite::increment_height() {
+  try {
+    SQLite::Transaction transaction{*db};
+    LOG_PRINT_L3("BlockchainDB_SQLITE::" << __func__);
+    this->height++;
+    SQLite::Statement update_height{*db,
+      "UPDATE batch_sn_info SET height = ?"};
```

Comment:
I think that `UPDATE batch_sn_info SET height = height + 1`  would work here if you are always just incrementing the value by 1.

Reply:
Yeah 100%, using the parameter is redundant in hindsight. 

What are your thoughts on calling the update here? This will get called after every block is added and im concerned it will be too slow, maybe updating the database height record in the destructor of my SQLite struct? Although that possibly will cause sync issues on an unexpected crash

Reply:
you could probably recompute expected height on startup to mitigate inconsistent state from crashes.